src/Dfc.CourseDirectory.WebV2/TLevelsBetaCanaryUserFeatureFlagProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+using Dfc.CourseDirectory.Core;
+using Dfc.CourseDirectory.WebV2.Security;
+
+namespace Dfc.CourseDirectory.WebV2
+{
+    public class TLevelsBetaCanaryUserFeatureFlagProvider : CanaryUserFeatureFlagProvider
+    {
+        public TLevelsBetaCanaryUserFeatureFlagProvider(IFeatureFlagProvider innerFeatureFlagProvider, ICurrentUserProvider currentUserProvider)
+            :base(innerFeatureFlagProvider, currentUserProvider)
+        {
+        }
+
+        protected override ISet<string> ResolveFeatureFlags(AuthenticatedUserInfo user, ISet<string> featureFlags)
+        {
+            if (!featureFlags.Contains(FeatureFlags.TLevels)
+                && featureFlags.Contains(FeatureFlags.TLevelsBeta)
+                && (user?.IsAdmin ?? false))
+            {
+                featureFlags.Add(FeatureFlags.TLevels);
```

Comment:
Given the signature of this method I'd expect it to return a new `ISet<T>` not mutate the argument. If mutating the existing instance is expected then I think this method should be `void`.

Reply:
Perhaps the `featureFlags` argument should be `IReadOnlyCollection<T>` (given we don't have `IReadOnlySet<T>` available to us yet).

Reply:
I see your point based on the concrete implementation, however I designed the API for this on the basis of producing an `ISet<string>` - the emitted result doesn't necessarily have to derive from the `ISet<string>`, it just so happens to do so in this circumstance.

> Perhaps the featureFlags argument should be IReadOnlyCollection<T> (given we don't have IReadOnlySet<T> available to us yet).

That might be a reasonable approach, I'll look into it.